$OpenBSD$

all of OpenBSD, FreeBSD, Apple, and NetBSD need to include sys/param.h before sys/sysctl.h (cross-checked using their respective web interfaces)

On NetBSD MAXLOGNAME is deprecated in favor of LOGIN_NAME_MAX

This avoids compilation errors like

/usr/include/sys/proc.h:65: error: 'MAXLOGNAME' was not declared in this scope
/usr/include/sys/proc.h:321: error: 'MAXCOMLEN' was not declared in this scope


--- boost/interprocess/detail/workaround.hpp.orig	Wed Nov 21 11:34:15 2012
+++ boost/interprocess/detail/workaround.hpp	Wed Nov 21 11:34:38 2012
@@ -110,6 +110,7 @@
 
    #if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
       #define BOOST_INTERPROCESS_BSD_DERIVATIVE
+      #include <sys/param.h>
       #include <sys/sysctl.h>
       #if defined(CTL_KERN) && defined (KERN_BOOTTIME)
          //#define BOOST_INTERPROCESS_HAS_KERNEL_BOOTTIME
